mmdet/core/bbox/assigners/uniform_assigner.py

```diff
@@ -0,0 +1,105 @@
+import torch
+
+from ..builder import BBOX_ASSIGNERS
+from ..iou_calculators import build_iou_calculator
+from ..transforms import bbox_xyxy_to_cxcywh
+from .assign_result import AssignResult
+from .base_assigner import BaseAssigner
+
+
+@BBOX_ASSIGNERS.register_module()
+class UniformAssigner(BaseAssigner):
+
+    def __init__(self,
+                 pos_ignore_thresh,
+                 neg_ignore_thresh,
+                 match_times=4,
+                 iou_calculator=dict(type='BboxOverlaps2D')):
+        self.match_times = match_times
+        self.pos_ignore_thresh = pos_ignore_thresh
+        self.neg_ignore_thresh = neg_ignore_thresh
+        self.iou_calculator = build_iou_calculator(iou_calculator)
+
+    def assign(self, bbox_pred, anchor, gt_bboxes, gt_labels, img_meta):
+        num_gts, num_bboxes = gt_bboxes.size(0), bbox_pred.size(0)
+
+        # 1. assign -1 by default
+        assigned_gt_inds = bbox_pred.new_full((num_bboxes, ),
+                                              0,
+                                              dtype=torch.long)
+        assigned_labels = bbox_pred.new_full((num_bboxes, ),
+                                             -1,
+                                             dtype=torch.long)
+        if num_gts == 0 or num_bboxes == 0:
+            # No ground truth or boxes, return empty assignment
+            if num_gts == 0:
+                # No ground truth, assign all to background
+                assigned_gt_inds[:] = 0
+            return AssignResult(
+                num_gts, assigned_gt_inds, None, labels=assigned_labels)
+
+        # Compute the L1 cost between boxes
+        # Note that we use anchors and predict boxes both
+        cost_bbox = torch.cdist(
+            bbox_xyxy_to_cxcywh(bbox_pred),
+            bbox_xyxy_to_cxcywh(gt_bboxes),
+            p=1)
+        cost_bbox_anchors = torch.cdist(
+            bbox_xyxy_to_cxcywh(anchor), bbox_xyxy_to_cxcywh(gt_bboxes), p=1)
+
+        # TODO: TOPK function has different results in cpu and cuda mode
+        C = cost_bbox.cpu()
```

Comment:
Why we move the cost to CPU and conduct `topk` on CPU?

Reply:
I refer to the open-source code, and then I found that the output results of CPU and GPU are different.

Reply:
If the difference does not affect the performance, we could use GPU for efficiency. No matter which choice, we should add comments to indicate the reason.